src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -40,6 +40,28 @@ public override NoPia.EmbeddedTypesManager EmbeddedTypesManagerOpt
         /// </summary>
         private Dictionary<FieldSymbol, NamedTypeSymbol> _fixedImplementationTypes;
 
+        private bool _needsGeneratedIsReadOnlyAttribute_IsFrozen;
+        private bool _needsGeneratedIsReadOnlyAttribute_Value;
+
+        /// <summary>
+        /// Returns a value indicating whether this builder has a symbol that needs IsReadOnlyAttribute to be generated during emit phase.
+        /// The value is set during during lowering the symbols that need that attribute, and is frozen on first trial to get it.
+        /// Freezing is needed to make sure that nothing tries to modify the value after the value is read.
+        /// </summary>
+        internal bool NeedsGeneratedIsReadOnlyAttribute
+        {
+            get
+            {
+                _needsGeneratedIsReadOnlyAttribute_IsFrozen = true;
+                return Compilation.NeedsGeneratedIsReadOnlyAttribute || _needsGeneratedIsReadOnlyAttribute_Value;
+            }
+            private set
```

Comment:
>private set [](start = 12, length = 11)

We probably never set this to false, therefore, I believe it is not appropriate to have setter for this property. Instead we should have a method that takes no arguments, named something like RecordNeedForGeneratedIsReadOnlyAttribute.